de.metas.business/src/main/java-legacy/org/compiere/process/BOMValidate.java

```diff
@@ -93,194 +81,160 @@ else if (I_PP_Product_BOM.Table_Name.equals(tableName))
-	}	//	validateProduct
+	}	// validateProduct
 
 
-	/**
-	 * 	Validate Old BOM Product structure
-	 *	@param product product
-	 *	@return true if valid
-	 */
-	private boolean validateOldProduct (MProduct product)
+	private void saveVerifiedProduct(@NonNull final I_M_Product product, final boolean isVerified)
 	{
-		if (!product.isBOM())
-			return true;
-		//
-		if (m_products.contains(product))
-		{
-			log.warn(m_product.getName() + " recursively includes " + product.getName());
-			return false;
-		}
-		m_products.add(product);
-		log.debug(product.getName());
-		//
-		MProductBOM[] productsBOMs = MProductBOM.getBOMLines(product);
-		for (int i = 0; i < productsBOMs.length; i++)
-		{
-			MProductBOM productsBOM = productsBOMs[i];
-			MProduct pp = new MProduct(getCtx(), productsBOM.getM_ProductBOM_ID(), get_TrxName());
-			if (!pp.isBOM())
-				log.trace(pp.getName());
-			else if (!validateOldProduct(pp))
-				return false;
-		}
-		return true;
-	}	//	validateOldProduct
+		product.setIsVerified(isVerified);
+		save(product);
+	}
 
 	/**
-	 * 	Validate BOM
-	 *	@param bom bom
-	 *	@return true if valid
+	 * Validate BOM
+	 *
+	 * @param bom bom
+	 * @return true if valid
 	 */
-	private boolean validateBOM (MBOM bom)
+	private boolean validateBOM(@NonNull final MBOM bom)
 	{
-		MBOMProduct[] BOMproducts = MBOMProduct.getOfBOM(bom);
+		final MBOMProduct[] BOMproducts = MBOMProduct.getOfBOM(bom);
 		for (int i = 0; i < BOMproducts.length; i++)
 		{
-			MBOMProduct BOMproduct = BOMproducts[i];
-			MProduct pp = new MProduct(getCtx(), BOMproduct.getM_BOMProduct_ID(), get_TrxName());
+			final MBOMProduct BOMproduct = BOMproducts[i];
+			final I_M_Product pp = InterfaceWrapperHelper.create(getCtx(), BOMproduct.getM_BOMProduct_ID(), I_M_Product.class, get_TrxName());
 			if (pp.isBOM())
+			{
 				return validateProduct(pp, bom.getBOMType(), bom.getBOMUse());
+			}
 		}
 		return true;
-	}	//	validateBOM
+	}
 
 	/**
-	 * 	Validate Product
-	 *	@param product product
-	 *	@param BOMType type
-	 *	@param BOMUse use
-	 *	@return true if valid
+	 * Validate Product
+	 *
+	 * @param product product
+	 * @param BOMType type
+	 * @param BOMUse use
+	 * @return true if valid
 	 */
-	private boolean validateProduct (MProduct product, String BOMType, String BOMUse)
+	private boolean validateProduct(@NonNull final I_M_Product product, String BOMType, String BOMUse)
 	{
 		if (!product.isBOM())
-			return true;
-		//
-		String restriction = "BOMType='" + BOMType + "' AND BOMUse='" + BOMUse + "'";
-		MBOM[] boms = MBOM.getOfProduct(getCtx(), p_M_Product_ID, get_TrxName(),
-			restriction);
-		if (boms.length != 1)
 		{
-			log.warn(restriction + " - Length=" + boms.length);
-			return false;
+			return true;
 		}
-		if (m_products.contains(product))
+
+		final String restriction = "BOMType='" + BOMType + "' AND BOMUse='" + BOMUse + "'";
+		final MBOM[] boms = MBOM.getOfProduct(getCtx(), p_M_Product_ID, get_TrxName(), restriction);
```

Comment:
drop this too... and drop any reference to old BOMs.
we are not using them at all.